pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -131,10 +131,26 @@ func GetPodConditionFromList(conditions []v1.PodCondition, conditionType v1.PodC
 	return -1, nil
 }
 
+// GetPodIPs returns the pod's IP addresses of given pod
+// Addresses should be fetched from Status.PodIPs if existing valuse,
+// otherwise, from Status.PodIP
+func GetPodIPs(pod *v1.Pod) []string {
+	var podIPs []string
+	if len(pod.Status.PodIPs) > 0 {
+		for _, itemIP := range pod.Status.PodIPs {
+			podIPs = append(podIPs, itemIP.IP)
+		}
+	} else {
+		podIPs = append(podIPs, pod.Status.PodIP)
+	}
+	return podIPs
+}
+
 func (pc *PodCache) labelFilter(old, cur *v1.Pod) bool {
 	// If labels updated, trigger proxy push
 	if cur.Status.PodIP != "" && !maps.Equal(old.Labels, cur.Labels) {
```

Comment:
checking `cur.Status.PodIP` then fetching from cur.Status.PodIPs seems wrong

Reply:
Hi @howardjohn, this cause many errors in testing due to these tests just specify the value for `cur.Status.PodIP` instead of `cur.Status.PodIPs`, fixing